### PR TITLE
OCPNODE-2281: Support multiple icsps in migrate icsp

### PR DESCRIPTION
**Pull request by QiWang19**
`oc adm migrate icsp` parses the icsp file using the separator to get all the icsp manifests and convert to single idms file.

### CODE REVIEW

**Comment by QiWang19** [comment]
@ardaguclu could you take a look?

File: pkg/cli/admin/migrate/icsp/icsp.go [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
 	"math/rand"
 	"os"
 	"path/filepath"
 
 	"github.com/spf13/cobra"
 	"gopkg.in/yaml.v3"
 
+	apicfgv1 "github.com/openshift/api/config/v1"
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	operatorv1alpha1scheme "github.com/openshift/client-go/operator/clientset/versioned/scheme"
```
Comment (ardaguclu) [inline pkg/cli/admin/migrate/icsp/icsp.go:17]: nit: separation of openshift group and k8s group

File: pkg/cli/admin/migrate/icsp/icsp.go [hunk 2]
```diff
@@ -138,60 +141,82 @@ func (o *MigrateICSPOptions) ensureDirectoryViable() error {
...
+			return []operatorv1alpha1.ImageContentSourcePolicy{}, "", fmt.Errorf("error decoding ImageContentSourcePolicy from %s: %v", icspFile, err)
+		}
+		icsp, ok := icspObj.(*operatorv1alpha1.ImageContentSourcePolicy)
+		if !ok {
+			return []operatorv1alpha1.ImageContentSourcePolicy{}, "", fmt.Errorf("could not decode ImageContentSourcePolicy from %s", icspFile)
+		}
+		icsps = append(icsps, *icsp)
 	}
-	return *icsp, icsp.Name, nil
+	return icsps, icsps[0].Name, nil
```
Comment (ardaguclu) [inline pkg/cli/admin/migrate/icsp/icsp.go:221]: Who is preventing the panic in here when `icsps` is empty?

Comment (QiWang19) [reply pkg/cli/admin/migrate/icsp/icsp.go:222]: In theory icsps is not empty when it returns from this line.
If the input file is empty, the "no data error" catches this error earlier https://github.com/openshift/oc/blob/7dfe72fee0e8d1ff8c4b7c82530311fd0b232e9f/pkg/cli/admin/migrate/icsp/icsp.go#L198-L200
If the input file has data the is not in the icsp scheme, the runtime decode can return the error https://github.com/openshift/oc/blob/7dfe72fee0e8d1ff8c4b7c82530311fd0b232e9f/pkg/cli/admin/migrate/icsp/icsp.go#L212-L215

**Pull request merged**